src/game/engine/sndctrl.cpp

```diff
@@ -0,0 +1,220 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "sndctrl.h"
+#include "callback.h"
+#include "gameloop.h"
+#include "gameoptions.h"
+#include "gbuffer.h"
+#include "mouse.h"
+#include "musiclist.h"
+#include "session.h"
+#include "stdio.h"
```

Comment:
This should be included as `#include <cstdio>` and any functions you use from it brought into the global namespace such as `using std::snprintf;`.